### DISCUSSION TITLE
How can I make it so that when I launch a program which takes a while to open, it opens on the one I originally asked it not - not where I am now?

### DISCUSSION THREAD

**Discussion by levihb** [i3/i3#5717] [Q&A]
This is always frustrating E.g. I open a window workspace, run gimp. It starts its load screen etc, then that disappears and the full window opens. Or other programs don't give you the progress bar, instead they just take longer to open.

What's annoying is I go and launch gimp, then of course I jump back to my previous workspace and carry on doing other things until it loads. But when it loads the window opens on the current workspace!!

It's really annoying and happens with so many applications. It seems an application always has to open where you currently are.

I'm not sure if this is an X thing? Or is it an i3 thing? Furthermore if it's an X thing, is there anyway to change it, or any workarounds for i3?

**Comment by twodogsdave** [comment]
this is how I do it..
bindsym $ALT+Control+n 			exec --no-startup-id i3-msg workspace 6; exec $music_player
bindsym $MOD+Shift+Control+o 	exec --no-startup-id i3-msg workspace 5; exec gamemoderun openspades

**Comment by orestisfl** [comment]
This is because of the [startup protocol](https://www.freedesktop.org/wiki/Software/startup-notification/). Using `exec --no-startup-id` should ignore the protocol.

E.g. test with:
```
i3-msg 'exec --no-startup-id <executable>, workspace 3'
```

**Comment by mikeredev** [comment]
had a similar issue, and I didn't necessarily want to give every program its own keybinding/workspace, so I ended up using `for_window [class=KeePass2|Gimp*] move to workspace $ws0` to make ws0 a kind of scratch workspace for apps I use frequently but rarely at the same time
